coursier.bzl

```diff
@@ -225,10 +232,57 @@ def generate_imports(repository_ctx, dep_tree, srcs_dep_tree = None):
             all_imports.append("alias(\n\tname = \"%s\",\n\tactual = \"%s\",\n)" % (versionless_target_alias_label, target_label))
 
         elif artifact_path == None:
-            fail("The artifact for " +
-                 artifact["coord"] +
-                 " was not downloaded. Perhaps the packaging type is not one of: jar, aar, bundle?\n" +
-                 "Parsed artifact data:" + repr(artifact))
+            # Possible reasons that the artifact_path is None:
+            #
+            # https://github.com/bazelbuild/rules_jvm_external/issues/70
+            # https://github.com/bazelbuild/rules_jvm_external/issues/74
+
+            # Get the reverse deps of the missing artifact.
+            reverse_deps = []
+
+            # For all potential reverse dep artifacts,
```

Comment:
Can you factor this out into a separate method?

Reply:
Done.